packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.test.tsx

```diff
@@ -0,0 +1,233 @@
+                        queryClient.getQueryState(['/customers', CUSTOMER_ID, '/addresses'])
+                            ?.isInvalidated
+                    ).toBeTruthy()
+                    expect(
+                        queryClient.getQueryState(['/customers', '/external-profile'])
+                            ?.isInvalidated
+                    ).toBeTruthy()
+                })
+            },
+            {
+                name: 'error',
+                assertions: withMocks(async () => {
+                    renderWithProviders(<CustomerMutationComponent />)
+                    await waitFor(() => screen.getByText(/kobe@test.com/))
+                    await waitFor(() =>
+                        screen.getByRole('button', {
+                            name: /update customer/i
+                        })
+                    )
+
+                    // CACHE Pre-populate cache with the query keys we invalidate on success
+                    queryClient.setQueryData(
+                        ['/customers', CUSTOMER_ID, '/payment-instruments'],
+                        {}
+                    )
+                    queryClient.setQueryData(['/customers', CUSTOMER_ID, '/addresses'], {})
+                    queryClient.setQueryData(['/customers', '/external-profile'], {})
+
+                    expect(
+                        queryClient.getQueryState([
+                            '/customers',
+                            CUSTOMER_ID,
+                            '/payment-instruments'
+                        ])?.isInvalidated
+                    ).toBeFalsy()
+                    expect(
+                        queryClient.getQueryState(['/customers', CUSTOMER_ID, '/addresses'])
+                            ?.isInvalidated
+                    ).toBeFalsy()
+                    expect(
+                        queryClient.getQueryState(['/customers', '/external-profile'])
+                            ?.isInvalidated
+                    ).toBeFalsy()
+
+                    // Mocking the server request
+                    nock('http://localhost:3000')
+                        .patch((uri) => {
+                            return uri.includes('/customer/shopper-customers/')
+                        })
+                        .reply(500)
+
+                    const button = screen.getByRole('button', {
+                        name: /update customer/i
+                    })
+
+                    fireEvent.click(button)
+                    await waitFor(() => screen.getByText(/error/i))
+                    expect(screen.getByText(/error/i)).toBeInTheDocument()
+
+                    // CACHE assert invalidated query keys
+                    expect(
+                        queryClient.getQueryState([
+                            '/customers',
+                            CUSTOMER_ID,
+                            '/payment-instruments'
+                        ])?.isInvalidated
+                    ).toBeFalsy()
```

Comment:
Do we need to test the error scenario? It feels like we can assume confidently that React Query will not run our `onSuccess` callback if the mutation has an error. And thus, the queries will not be invalidated.